internal/nginx/config/variable_names.go

```diff
@@ -9,3 +9,14 @@ import (
 func convertStringToSafeVariableName(s string) string {
 	return strings.ReplaceAll(s, "-", "_")
 }
+
+// Generate the variable name for a proxy add header map.
+func generateAddHeaderMapVariableName(name string) string {
+	underscoreName := strings.ToLower(convertStringToSafeVariableName(name))
+	// Ensure the length of this var is not too long - to prevent cases like:
+	// "could not build map_hash, you should increase map_hash_bucket_size: 64"
+	if len(underscoreName) <= 35 {
+		return underscoreName + "_header_var"
+	}
+	return underscoreName[:35] + "_header_var"
```

Comment:
Could this possibly collide with a shorter name that shares the first 35 characters? I wonder if it's worth it to add a special delimiter in this case to prevent collisions.

Reply:
another option is too increase `map_hash_bucket_size:`
another option is to generate names like $header_add_0, $header_add_1, $header_add_2 ...

Reply:
I'm not sure what a special delimiter could be in this case because IIRC an NGINX var can only contain alphanumeric characters and underscores? 

I have removed truncation and increased `variables_hash_bucket_size 128;` (which is actually the error I was hitting first on start-up. I'm not getting a `map_hash_bucket_size` error). This covers all cases of header name length > 45 characters, which is max size the proxy_set_header field can be before we get the `could not build proxy_headers_hash, you should increase proxy_headers_hash_bucket_size: 64` error. Thoughts?